src/modules/app/containers/__tests__/NavigatorContainer.spec.js

```diff
@@ -0,0 +1,97 @@
+// @flow
+
+import TestRenderer from 'react-test-renderer'
+import { Provider } from 'react-redux'
+import Navigator from '../../components/Navigator'
+import configureMockStore from 'redux-mock-store'
+import { render } from '@testing-library/react-native'
+import * as React from 'react'
+import NavigatorContainer from '../NavigatorContainer'
+import waitForExpect from 'wait-for-expect'
+import AppSettings from '../../../settings/AppSettings'
+import AsyncStorage from '@react-native-community/async-storage'
+
+jest.mock('../../../../routes/landing/containers/LandingContainer', () => {
+  const Text = require('react-native').Text
+  return () => <Text>Landing</Text>
+})
+jest.mock('../../../../routes/dashboard/containers/DashboardContainer', () => {
+  const Text = require('react-native').Text
+  return () => <Text>Dashboard</Text>
+})
+jest.mock('../../../layout/containers/HeaderContainer', () => {
+  const Text = require('react-native').Text
+  return () => <Text>Header</Text>
+})
+jest.mock('@react-native-community/async-storage')
+jest.mock('react-native-share')
+
+const mockStore = configureMockStore()
+
+describe('NavigatorContainer', () => {
+  beforeEach(() => {
+    AsyncStorage.clear()
+    jest.resetModules()
+  })
+
+  it('should pass fetchCategory to Navigator', () => {
+    jest.doMock('../../components/Navigator', () => () => null)
```

Comment:
I think this mock is never actually used. The `Navigator` corresponds to the instance which was imported at the top of the file. Also because `NavigatorContainer` and therefore the Navigator  is imported before the doMock there is no way that mock can be used.

The solution is to import the NavigatorContainer and Navigator using `require`.

Reply:
Yup, thanks for discovering! Fixed it.